pkg/k8s/watchers/pod.go

```diff
@@ -702,9 +720,22 @@ func (k *K8sWatcher) updatePodHostData(oldPod, newPod *slim_corev1.Pod, oldPodIP
 		}
 	}()
 
-	err := k.upsertHostPortMapping(oldPod, newPod, oldPodIPs, newPodIPs)
-	if err != nil {
-		return fmt.Errorf("cannot upsert hostPort for PodIPs: %s", newPodIPs)
+	specEqual := oldPod != nil && newPod.Spec.DeepEqual(&oldPod.Spec)
+	ipSliceEqual := oldPodIPs != nil && oldPodIPs.DeepEqual(&newPodIPs)
+	hostIPEqual := oldPod != nil && newPod.Status.HostIP != oldPod.Status.HostIP
+
+	// only upsert HostPort Mapping if spec or ip slice is different
+	if !specEqual || !ipSliceEqual {
+		err := k.upsertHostPortMapping(oldPod, newPod, oldPodIPs, newPodIPs)
+		if err != nil {
+			return fmt.Errorf("cannot upsert hostPort for PodIPs: %s", newPodIPs)
```

Comment:
Just wondering, will this error appear only in the agent long or it does get surfaced as an event on the pod either directly or via CNI?

Reply:
It only shows up in the cilium-agent logs. It's a similar behavior as kube-proxy.

Reply:
https://github.com/cilium/cilium/pull/16190#discussion_r638509416
(2) is not addressed still?